server.js

```diff
@@ -32,6 +32,92 @@ app.post('/api/savings', async (req, res) => {
     }
 });
 
+
+//Api to count how many days to the next public holiday
+app.get('/api/showCalendar', async function(req, res){
+    try {
+        const t = await calculateDateToNextHoliday()
+        console.log(t);
+        res.send(t);
+    } catch (error) {
+        console.log(error);
+        res.status(500).send('Internal server error');
+    }
+});
+
+
+//get holidays from google calandar API, however the return JSON not only contains current year holidays 
+async function getHoliday(){
+    const BASE_CALENDAR_URL = "https://www.googleapis.com/calendar/v3/calendars";
+    const BASE_CALENDAR_ID_FOR_PUBLIC_HOLIDAY =
+      "holiday@group.v.calendar.google.com"; // Calendar Id. This is public but apparently not documented anywhere officialy.
+    const API_KEY = "AIzaSyBg0IiUjXv79hXM_V5aYxKAd6erKPPYvCU";
+    const CALENDAR_REGION = "en.irish"; // This variable refers to region whose holidays do we need to fetch
+
+    const url = `${BASE_CALENDAR_URL}/${CALENDAR_REGION}%23${BASE_CALENDAR_ID_FOR_PUBLIC_HOLIDAY}/events?key=${API_KEY}`
+
+    const response = await fetch(url);
+    const data = await response.json();
+    const holidays = data.items;
+    return holidays;
+}
+
+//filter the holidays to get only current year holiday
+async function filterCurrentYearHoliday(){
+    const holidays = await getHoliday();
+    const currentYear = new Date().getFullYear()  // returns the current year
+   
+    var holidaysAndDate = [];   //filter the result to make it only contains holiday name and time
+    for(var i in holidays){
+        var singleItem = holidays[i];
+        var summay = singleItem.summary;
+        var time = singleItem.start.date;
+        var timeToCompare = time.split('-')[0];
+
+        if(timeToCompare == currentYear){
+            holidaysAndDate.push({ 
+                "summary" : summay,
+                "time"  : time,
+            });
+            if(summay == "New Year's Eve"){
```

Comment:
is this for checking if it is next year already?

Reply:
no only currently year